target.go

```diff
@@ -186,25 +225,50 @@ func (s *targetServer) obliviousQueryHandler(w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	packedResponse, err := s.resolveQuery(query)
+	queryParseAndDecryptionCompleteTime := time.Now().UnixNano()
+	timestamp.TargetQueryDecryptionTime = queryParseAndDecryptionCompleteTime
+
+	resolverChosen := s.resolver[chosenResolver]
+	packedResponse, err := s.resolveQueryUsingNS(query, resolverChosen)
 	if err != nil {
 		log.Println("Failed resolving DNS query:", err)
 		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 		return
 	}
 
+	queryResolutionCompleteTime := time.Now().UnixNano()
+	timestamp.TargetQueryResolutionTime = queryResolutionCompleteTime
+
 	obliviousResponse, err := s.createObliviousResponseForQuery(obliviousQuery, packedResponse)
 	if err != nil {
 		log.Println("Failed creating DNS oblivious DNS response:", err)
+		timestamp.TargetAnswerEncryptionTime = 0
+		timestamp.EndTime = 0
+		exp.Timestamp = timestamp
+		exp.Status = false
+		exp.Resolver = ""
+		go s.telemetryClient.streamDataToElastic([]string{exp.serialize()})
```

Comment:
Can we abstract over Elasticsearch? For example, we could simply name this function `submit`, and then the specific telemetry client could determine how and where to send the data.

Reply:
Also, calling `streamDataToElastic` once every query could be costly. Should we batch them? For example, we could just do:

```suggestion
s.telemetryClient.submit(exp.serialize())
```

And internally, for the Elasticsearch telemetry client, that would queue up strings to log, and send them when done.

Reply:
I'll do this in parallel with https://github.com/chris-wood/odoh-client/issues/17 when its being tackled on the client.